### PR TITLE
8259943: FileDescriptor.close0 does not handle EINTR

### PR DESCRIPTION
**Pull request by bplb**
Please review this small change to handle `EINTR` from `close()` in `fileDescriptorClose()`. The function `handleGetLength()` is also changed to handle `EINTR` from `fstat64()` to match other uses of `fstat64()` in the file.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8259943](https://bugs.openjdk.java.net/browse/JDK-8259943): FileDescriptor.close0 does not handle EINTR

### Reviewers
 * [Naoto Sato](https://openjdk.java.net/census#naoto) (@naotoj - **Reviewer**) ⚠️ Review applies to 047dac0b35ecdf70fb22623d0ec15cd8e0fbc4c1
 * [Alan Bateman](https://openjdk.java.net/census#alanb) (@AlanBateman - **Reviewer**)

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/2173/head:pull/2173`
`$ git checkout pull/2173`

### CODE REVIEW

**Review by naotoj — APPROVED** [review]
Looks good, Brian.

File: src/java.base/unix/native/libjava/io_util_md.c [hunk 1]
```diff
@@ -162,8 +162,12 @@ fileDescriptorClose(JNIEnv *env, jobject this)
             dup2(devnull, fd);
             close(devnull);
         }
-    } else if (close(fd) == -1) {
-        JNU_ThrowIOExceptionWithLastError(env, "close failed");
+    } else {
+        int result;
+        RESTARTABLE(close(fd), result);
```
Comment (AlanBateman) [inline src/java.base/unix/native/libjava/io_util_md.c:167]: close is not restartable (except AIX apparently). This means you have to explicitly handle EINTR, see UnixNativeDispatcher.close0 for an example.

Comment (bplb) [reply src/java.base/unix/native/libjava/io_util_md.c:167]: I had it that way before creating the PR; will revert.

File: src/java.base/unix/native/libjava/io_util_md.c [hunk 2]
```diff
@@ -162,8 +162,17 @@ fileDescriptorClose(JNIEnv *env, jobject this)
...
-        JNU_ThrowIOExceptionWithLastError(env, "close failed");
+    } else {
+        int result;
+#if defined(_AIX)
+        /* AIX allows close to be restarted after EINTR */
+        RESTARTABLE(close(fd), result);
+#else
+        result = close(fd);
+#endif
+        if (result == -1) {
```
Comment (AlanBateman) [inline src/java.base/unix/native/libjava/io_util_md.c:173]: This needs to be `if (result == -1 && errno != EINTR)`

Comment (bplb) [reply src/java.base/unix/native/libjava/io_util_md.c]: Yes, I already caught and fixed that gaffe.

**Review by AlanBateman — APPROVED** [review]

**Pull request closed without merging**